optaplanner-integration/optaplanner-spring-boot-example/src/main/java/org/optaplanner/spring/boot/example/poc/impl/solver/DefaultSolverManager.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.spring.boot.example.poc.impl.solver;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.spring.boot.example.poc.api.solver.SolverManager;
+
+/**
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ */
+// THIS IS JUST A PROOF OF CONCEPT IN THE EXAMPLE FOR 7.30.0.Final. THIS CLASS WILL BECOME OBSOLETE VERY SOON.
+// TODO Clean this up and move this class into optaplanner-core
+public class DefaultSolverManager<Solution_> implements SolverManager<Solution_> {
+
+    private final SolverFactory<Solution_> solverFactory;
+    private ExecutorService executorService;
+
+    public DefaultSolverManager(SolverFactory<Solution_> solverFactory) {
+        this.solverFactory = solverFactory;
+        validateSolverFactory();
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    private void validateSolverFactory() {
+        solverFactory.buildSolver();
+    }
+
+    @Override
+    public DefaultSolverFuture solve(Solution_ planningProblem,
+            Consumer<Solution_> bestSolutionConsumer) {
+        Solver<Solution_> solver = solverFactory.buildSolver();
+        // TODO consumption should happen on different thread than solver thread, doing skipAhead and throttling
+        solver.addEventListener(event -> bestSolutionConsumer.accept(event.getNewBestSolution()));
+        DefaultSolverFuture solverFuture = new DefaultSolverFuture<>(solver);
+        executorService.submit(() -> {
+            try {
+                solver.solve(planningProblem);
+            } catch (Exception e) {
+                e.printStackTrace(); // TODO generated
```

Comment:
This definitely isn't production-ready.

Reply:
Hmm - good point. We can't release example code like this. Maybe we should mask out the example?

Reply:
I'm not sure what you're suggesting. How would masking out look?

Reply:
Fixed, now an exceptionHandler is supported